x/auth/client/cli/sign.go

```diff
@@ -122,35 +128,70 @@ func makeSignCmd(cdc *amino.Codec) func(cmd *cobra.Command, args []string) error
 		if err != nil {
 			return err
 		}
+
 		defer fp.Close()
 		fmt.Fprintf(fp, "%s\n", json)
+
 		return
 	}
 }
 
-func printSignatures(stdTx auth.StdTx) bool {
+// printAndValidateSigs will validate the signatures of a given transaction over
+// its expected signers. In addition, if offline has not been supplied, the
+// signature is verified over the transaction sign bytes.
+func printAndValidateSigs(
+	cliCtx context.CLIContext, chainID string, stdTx auth.StdTx, offline bool,
+) bool {
+
 	fmt.Println("Signers:")
+
 	signers := stdTx.GetSigners()
 	for i, signer := range signers {
 		fmt.Printf(" %v: %v\n", i, signer.String())
 	}
 
+	success := true
 	sigs := stdTx.GetSignatures()
+
 	fmt.Println("")
 	fmt.Println("Signatures:")
-	success := true
+
 	if len(sigs) != len(signers) {
 		success = false
 	}
-	for i, sig := range stdTx.GetSignatures() {
+
+	for i, sig := range sigs {
 		sigAddr := sdk.AccAddress(sig.Address())
 		sigSanity := "OK"
+
 		if i >= len(signers) || !sigAddr.Equals(signers[i]) {
-			sigSanity = fmt.Sprintf("ERROR: signature %d does not match its respective signer", i)
+			sigSanity = "ERROR: signature does not match its respective signer"
 			success = false
 		}
+
+		// Validate the actual signature over the transaction bytes since we can
+		// reach out to a full node to query accounts.
+		if !offline {
```

Comment:
As per our conversation in Slack, I'd propose to short circuit if signer and signature don't match